### PR TITLE
Fix completions within select clause context

### LINKED ISSUE #33934: On conflict clause not suggested after select clause in query expression context
**Description:**
onconflict

As depicted above on conflict keyword pair is not suggested at the depicted cursor position.

```ballerina
function findTop10GoldMedalsToAthletesRatio() returns error? {
    map<int> athletesByCountry = check findNumOfAthletesBycountry();
    stream<MedalStats, error?> medalStats = check getMedalStats();

    MyError onConflictError = error("Conflict");

    var ratios = from [string,int] entry in athletesByCountry.entries() 
    join var stats in medalStats on entry[0] equals stats.country 
    let float ratio = (<float>stats.gold/<float>entry[1]) * 100 order by ratio descending 
    limit 10
    select  {
        country: entry[0],
        ratio
    } o<cursor> ;

    io:println("Top 10 countries with gold medals/athletes ratio");
    
    foreach var ratio in ratios {
        io:println("\t", ratio.country, "\t", float:round(ratio.ratio), "%");
    }
}
```

Further, the sorting order in the on-conflict-clause context can be improved to order the expressions with `error` subtype at the top.

onConflict

### PR DESCRIPTION
**Pull request by nadeeshaan**
## Purpose
Fixes #33934 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by mohanvive — APPROVED** [review]

**Comment by mohanvive** [comment]
One thing to highlight, on-conflict clause is used only when creating a table typed value (out put of the query expression is table) using query expressions.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/OnConflictClauseNodeContext.java [hunk 1]
```diff
@@ -57,12 +65,37 @@ public OnConflictClauseNodeContext() {
...
+                }
+            }
+            String sortText = SortingUtil.genSortText(rank)
+                    + SortingUtil.genSortText(SortingUtil.toRank(context, lsCItem));
+            lsCItem.getCompletionItem().setSortText(sortText);
+        }
+    }
+
+    private boolean isVariableOfErrorType(LSCompletionItem lsCItem) {
+        Predicate<Symbol> symbolPredicate = CommonUtil.getVariableFilterPredicate();
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/OnConflictClauseNodeContext.java:96]: How about function call expressions of error type? We haven't considered them here.

**Review by malinthar — CHANGES_REQUESTED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/OnConflictClauseNodeContext.java (hunk 1 above)
Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/OnConflictClauseNodeContext.java:98]: Done

**Review by malinthar — APPROVED** [review]
LGTM!

**Pull request merged**